x-pack/plugins/monitoring/server/lib/elasticsearch/ccr.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+
+export function handleResponse(response) {
+  let enabled = true;
+  const sources = ['persistent', 'transient', 'defaults'];
+  for (const source of sources) {
+    const ccrEnabled = get(response[source], 'xpack.monitoring.ccr.enabled');
+    if (ccrEnabled === false) {
```

Comment:
I haven't been able to test this until https://github.com/elastic/elasticsearch/pull/37432 is resolved as I can't start ES with ccr disabled.

Reply:
Two things here:

1. The precedence order of settings in ES is transient > persistent > defaults. So if a setting is set as a transient setting, no matter what it's value, that's the value we should be looking at.

2. In the case of this particular setting `xpack.monitoring.ccr.enabled`, you'll never find it under `transient` or `persistent` since it's not a dynamically updatable setting.

So I'd suggest doing one of two things to this function:

1. Either make it generic enough to handle a response from `GET _cluster/settings`, following the correct precedence order, and then call it with the `xpack.monitoring.ccr.enabled` setting, OR

2. Make it specific to the `xpack.monitoring.ccr.enabled` setting but only look at the value under `defaults`.

Reply:
Good point. I updated it to only look in the defaults and I actually mistyped the string in the handleResponse method.